server/chess.py

```diff
@@ -100,16 +100,17 @@ def hypothetical_check(
                 typing.Tuple[models.Piece, int, int], ...
             ]) -> bool:
         """Check if a series of moves would put a side in check."""
-        if self.hypothetical_moves is None:
+        if moves is None:
             raise RuntimeError('Checkmate detection recursion detected.')
         self.hypothetical_moves = moves    # self.get_piece will observe this
+        """***Note that the hypothetical moves are not yet being used***"""
```

Comment:
```suggestion
        # FIXME: Observe hypothetical moves in get_piece
```
Triple quotes should not be used for comments, the exception is docstrings which I suppose are comments.